File: app/src/main/java/com/example/multidagger/di/AppModule.java
package com.example.multidagger.di;

import android.app.Application;

import android.content.Context;
import android.content.SharedPreferences;

import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
abstract class AppModule {

    @Singleton
    @Provides
    static SharedPreferences providesSharedPreference(Application application) {
        return application.getSharedPreferences("test", Context.MODE_PRIVATE);
    }
}

File: app/src/main/kotlin/com/example/multidagger/di/AppModule.kt
package com.example.multidagger.di

import android.content.Context
import android.content.SharedPreferences

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent

@InstallIn(SingletonComponent::class)
@Module
internal object AppModule {

    @Provides
    fun providesSharedPreference(@ApplicationContext context: Context): SharedPreferences =
        context.getSharedPreferences("test", Context.MODE_PRIVATE)
}
